milterfrom.c

```diff
@@ -56,13 +56,15 @@ struct mlfiPriv {
 
 static unsigned long mta_caps = 0;
 
-// Function to extract addresses from the header/envelope fields.
-// If the field contains a < with a subsequent >, the inner part is used. If not, the whole header field is used. This allows matching "Max Mustermann <max.mustermann@example.invalid>" matching.
-const char *parse_address(const char *address, int *len)
+// Function to extract addresses from the header/envelope fields.  If the field
+// contains a < with a subsequent >, the inner part is used. If not, the whole
+// header field is used. This allows matching "Max Mustermann
+// <max.mustermann@example.invalid>" matching.
+const char *parse_address(const char *address, size_t *len)
 {
-	int inlen = strlen(address);
-	int pos_open = -1, pos_close = -1;
-	int i;
+	size_t inlen = strlen(address);
+	size_t pos_open = -1, pos_close = -1;
```

Comment:
probably we should use SIZE_MAX here and in the condition below, as the behavior is more obvious